src/plugins/signoff/sagas.js

```diff
@@ -110,57 +87,21 @@ export function* onCollectionRecordsRequest(getState, action) {
   yield put(SignoffActions.workflowInfo(collectionsInfo));
 }
 
-export function* fetchWorkflowInfo(source, preview, destination) {
-  // Obtain collections attributes for source, preview and destination.
+export async function fetchWorkflowInfo(source, preview, destination) {
   const client = getClient();
-  const fetchInfos = batch => {
-    batch
-      .bucket(source.bucket)
-      .collection(source.collection)
-      .getData();
-    batch
-      .bucket(preview.bucket)
-      .collection(preview.collection)
-      .getData();
-    batch
-      .bucket(destination.bucket)
-      .collection(destination.collection)
-      .getData();
-  };
-  const resp = yield call([client, client.batch], fetchInfos);
-  // Extract the `data` attribute from each response body, and defaults to
-  // empty {} if missing (eg. if preview or destination don't exist yet).
-  const respData = resp.map(({ status, body: { data = {} } }, index) => data);
-  const [sourceAttributes, previewAttributes, destinationAttributes] = respData;
-  // Fetch the source changes made since the last signature (since=0 means everything).
-  const lastSigned = String(destinationAttributes.last_modified || 0);
   const { bucket: bid, collection: cid } = source;
   const colClient = client.bucket(bid).collection(cid);
-  const { data: sourceChanges } = yield call(
-    [colClient, colClient.listRecords],
-    { since: lastSigned }
-  );
-  // Here, `lastUpdated` gives us the timestamp of the most recently changed record.
-  // Which can be different from `sourceAttributes.last_modified` since the collection
-  // attributes can be changed independently from the records.
-  let lastUpdated;
-  if (sourceChanges.length > 0) {
-    // Some changes exist. Easy, take latest timestamp.
-    lastUpdated = sourceChanges[0].last_modified;
-  } else {
-    // Empty changes list: either never signed, or not changed since last signature.
-    lastUpdated = lastSigned === "0" ? null : sourceAttributes.last_modified;
-  }
+
+  const sourceAttributes = await colClient.getData();
```

Comment:
Kind of a n00b question here, what's the difference between using `yield call` (as we were doing previously) and `await` (as we're doing now)? Was `yield call` an essential part of sagas or was it a workaround because JS didn't have async/await yet?

Reply:
`yield` is called from generators functions, which are an essential part of sagas. `yield call` is used to call `async` functions or other sagas/generators.

This `fetchWorkflowInfo` function was not a saga, just a util, and can thus be written as a simple async function.

I guess that it was initially written as a generator «by mistake» https://github.com/Kinto/kinto-admin/commit/7db498419739447df3dbb4ce7a68d6403c964bd8#diff-4690b38f7530de693841dfc00502ad16R69